src/besseli.jl

```diff
@@ -193,3 +194,83 @@ function besseli_large_orders_scaled(v, x::T) where T <: Union{Float32, Float64,
 
     return T(coef*Uk_poly_In(p, v, p2, T))
 end
+
+function _besseli_continued_fractions(nu, x::T) where T
+    S = promote_type(T, Float64)
+    xx = S(x)
+    knu, knum1 = up_recurrence(xx, besselk0(xx), besselk1(xx), nu)
+    (iszero(knu) || iszero(knum1)) && return throw(DomainError(x, "Overflow error"))
+    return 1 / (x * (knum1 + knu / steed(nu, x)))
+end
+
+function steed(n, x::T) where T
+    MaxIter = 1000
+    xinv = inv(x)
+    xinv2 = 2 * xinv
+    d = x / (n + n)
+    a = d
+    h = a
+    b = muladd(2, n, 2) * xinv
+    for _ in 1:MaxIter
+        d = inv(b + d)
+        a = muladd(b, d, -1) * a
+        h = h + a
+        b = b + xinv2
+        abs(a / h) <= eps(T) && break
+    end
+    return h
+end
+
+function besseli_large_argument(v, z::T) where T
+    MaxIter = 1000
+    a = exp(z / 2)
+    coef = a / sqrt(2 * T(pi) * z)
+    fv2 = 4 * v^2
+    term = one(T)
+    res = term
+    s = -term
+    for i in 1:MaxIter
+        i = T(i)
+        offset = muladd(2, i, -1)
+        term *= T(0.125) * muladd(offset, -offset, fv2) / (z * i)
+        res = muladd(term, s, res)
+        s = -s
+        abs(term) <= eps(T) && break
+    end
+    return res * coef * a
+end
+
+# power series definition of besseli
+# fast convergence for small arguments
+# for large orders and small arguments there is increased roundoff error ~ 1e-14
+# need to investigate further if this can be mitigated
+function besseli_small_arguments(v, z::T) where T
+    if v < 20
+        coef = (z / 2)^v / factorial(v)
+    else
+        coef = v*log(z / 2)
+        coef -= _loggam(v + 1)
+        coef = exp(coef)
+    end
+
+    MaxIter = 1000
+    out = one(T)
+    zz = z^2 / 4
+    a = one(T)
+    for k in 0:MaxIter
+        a *= zz / (k + 1) / (k + v + 1)
```

Comment:
`zz / ((k + 1)  * (k + v + 1))` should be faster.

Reply:
I think this block can just be replaced by
```julia
zz = z^2 / 4
    a = one(T)
    for k in 1:MaxIter
        a *= zz / (k * (k + v))
        out += a
        a <= eps(T) && break
    end
```